extensions/sql-database-projects/src/test/dialogs/createProjectFromDatabaseQuickpick.test.ts

```diff
@@ -0,0 +1,316 @@
+
+		//user chooses connection and database
+		sinon.stub(testUtils.vscodeMssqlIExtension.object, 'connect').resolves('testConnectionURI');
+		sinon.stub(testUtils.vscodeMssqlIExtension.object, 'listDatabases').withArgs(sinon.match.any).resolves(dbList);
+		let quickPickStub = sinon.stub(vscode.window, 'showQuickPick').resolves('Database' as any);
+		//user chooses project name
+		sinon.stub(vscode.window, 'showInputBox').resolves(projectFileName);
+		// user chooses a folder/file combination that already exists
+		quickPickStub.onSecondCall().resolves(testProjectFilePath as any);
+		//user chooses another folder when prompted again
+		quickPickStub.onThirdCall().resolves(path.join(projectFilePath, 'test') as any);
+		//user chooses to exit when prompted for folder structure
+		quickPickStub.onCall(3).resolves(undefined);
+
+		const model = await createProjectFromDatabaseQuickpick.createNewProjectFromDatabaseWithQuickpick(mockConnectionInfo);
+
+		await fs.rm(testProjectFilePath, { recursive: true });
+
+		//verify showQuickPick exited with undefined, since folder structure wasn't selected (resolved to undefined)
+		should.equal(model, undefined);
+	});
+
+	it('Should exit when include permissions is not selected', async function (): Promise<void> {
+		sinon.stub(utils, 'getVscodeMssqlApi').resolves(testUtils.vscodeMssqlIExtension.object);	//set vscode mssql extension api
+
+		let dbList: string[] = constants.systemDbs;
+		dbList.push('OtherDatabase');
+		dbList.push('Database');
+		dbList.push('OtherDatabase2');
+
+		//user chooses connection and database
+		sinon.stub(testUtils.vscodeMssqlIExtension.object, 'connect').resolves('testConnectionURI');
+		sinon.stub(testUtils.vscodeMssqlIExtension.object, 'listDatabases').withArgs(sinon.match.any).resolves(dbList);
+		let quickPickStub = sinon.stub(vscode.window, 'showQuickPick').resolves('Database' as any);
+		//user chooses project name
+		sinon.stub(vscode.window, 'showInputBox').resolves('TestProject');
+		// user chooses a folder
+		quickPickStub.onSecondCall().resolves(projectFilePath as any);
+		//user chooses Object type when prompted for folder structure
+		quickPickStub.onThirdCall().resolves(constants.objectType as any);
+		//user chooses to exit when prompted for include permissions
+		quickPickStub.onCall(3).resolves(undefined);
+
+		const model = await createProjectFromDatabaseQuickpick.createNewProjectFromDatabaseWithQuickpick(mockConnectionInfo);
+
+		//verify showQuickPick exited with undefined, since include permissions wasn't selected (resolved to undefined)
+		should.equal(model, undefined);
+	});
+
+	it('Should exit when sdk style project is not selected', async function (): Promise<void> {
+		sinon.stub(utils, 'getVscodeMssqlApi').resolves(testUtils.vscodeMssqlIExtension.object);	//set vscode mssql extension api
+
+		let dbList: string[] = constants.systemDbs;
+		dbList.push('OtherDatabase');
+		dbList.push('Database');
+		dbList.push('OtherDatabase2');
```

Comment:
could just have a const string array of dbs defined at the top of this file that gets used by all these tests